viadot/sources/databricks.py

```diff
@@ -0,0 +1,350 @@
+import time
+# from viadot.sources import Base
+class Databricks(Source):
+
+    defaults_key = "DEFAULTS"
+    env = ""
+    spark = None
+
+    def __init__(self,  env: str = "QA", *args, **kwargs):
+        self.env = env
+        org_id = local_config.get("DATABRICKS", {}).get(Databricks.defaults_key, {}).get("org_id")
+        port = local_config.get("DATABRICKS", {}).get(Databricks.defaults_key, {}).get("port")
+        print(org_id)
+        default_credentials = {
+            "spark.databricks.service.org_id": org_id,
+            "spark.databricks.service.port": port
+        }
+        self.credentials = default_credentials or local_config.get("DATABRICKS", {}).get(env)
+        self.connect()
+        
+    def _get_spark_session(self, env: str = None):
+        env = env or self.env
+        default_spark = SparkSession.builder.getOrCreate()
+        new_config = SparkConf()
+        # copy all the configuration values from the current Spark Context
+        default_config_vals = default_spark.sparkContext.getConf().getAll()
+        for (k, v) in default_config_vals:
+            new_config.set(k, v)
+        new_config.set("spark.databricks.service.clusterId", self.credentials.get("cluster_id"))
+        new_config.set("spark.databricks.service.port", self.credentials.get("port"))
+        # stop the spark session context in order to create a new one with the required cluster_id, else we
+        # will still use the current cluster_id for execution
+        default_spark.stop()
+        context = SparkContext(conf=new_config)
+        session = SparkSession(context)
+        new_spark = session.builder.config(conf=new_config).getOrCreate()
+        return new_spark
+
+    def set_custom_config(self, cluster_id:str, port: str = "15001"):
+        default_spark = SparkSession.builder.getOrCreate()
+        new_config = SparkConf()
+        # copy all the configuration values from the current Spark Context
+        default_config_vals = default_spark.sparkContext.getConf().getAll()
+        for (k, v) in default_config_vals:
+            new_config.set(k, v)
+        new_config.set("spark.databricks.service.clusterId", cluster_id)
+        new_config.set("spark.databricks.service.port", port)
+        # stop the spark session context in order to create a new one with the required cluster_id, else we
+        # will still use the current cluster_id for execution
+        default_spark.stop()
+        context = SparkContext(conf=new_config)
+        session = SparkSession(context)
+        new_spark = session.builder.config(conf=new_config).getOrCreate()
+        return new_spark
+
+    def connect(self):
+        if self.env == "QA":
+            self.spark = SparkSession.builder.getOrCreate()
+        else:
+            self.spark = self._get_spark_session(self.env)
+    
+    def to_df(self, list):
+        return pd.DataFrame(list)
+    
+    def to_json(sparkDf: pyspark.sql.dataframe.DataFrame):
```

Comment:
I would import this as `spark.DataFrame` for clarity. Then you have `pd.DataFrame` and `spark.DataFrame`.